firstvoices/backend/management/commands/copy_site.py

```diff
@@ -0,0 +1,581 @@
+            "videos": list(song.related_videos.all().values_list("id", flat=True)),
+        }
+
+        song.id = uuid.uuid4()
+        song.site = target_site
+        song.save()
+
+        for lyric in source_lyrics:
+            lyric.id = uuid.uuid4()
+            lyric.song = song
+            lyric.save()
+
+        copy_related_media(song, source_media, audio_map, image_map, video_map)
+        song.save()
+
+
+def copy_stories(source_site, target_site, audio_map, image_map, video_map):
+    stories = list(Story.objects.filter(site=source_site))
+    for story in stories:
+        source_pages = list(story.pages.all())
+        source_media = {
+            "audio": list(story.related_audio.all().values_list("id", flat=True)),
+            "images": list(story.related_images.all().values_list("id", flat=True)),
+            "videos": list(story.related_videos.all().values_list("id", flat=True)),
+        }
+
+        story.id = uuid.uuid4()
+        story.site = target_site
+        story.save()
+
+        copy_related_media(story, source_media, audio_map, image_map, video_map)
+        story.save()
+
+        for page in source_pages:
+            source_media = {
+                "audio": list(page.related_audio.all().values_list("id", flat=True)),
+                "images": list(page.related_images.all().values_list("id", flat=True)),
+                "videos": list(page.related_videos.all().values_list("id", flat=True)),
+            }
+            page.id = uuid.uuid4()
+            page.site = target_site
+            page.story = story
+            page.save()
+
+            copy_related_media(page, source_media, audio_map, image_map, video_map)
+            page.save()
+
+
+def copy_dictionary_entries_and_return_map(
+    source_site,
+    target_site,
+    category_map,
+    character_map,
+    audio_map,
+    image_map,
+    video_map,
+):
+    dictionary_entry_map = {}
+
+    dictionary_entries = list(
+        DictionaryEntry.objects.filter(site=source_site)
+        .annotate(related_entry_count=Count("related_dictionary_entries"))
+        .order_by("related_entry_count")
+        .distinct()
+    )
+    for entry in dictionary_entries:
+        source_categories = list(entry.categories.all())
+        source_related_characters = list(entry.related_characters.all())
+        source_related_entries = list(entry.related_dictionary_entries.all())
+        source_media = {
+            "audio": list(entry.related_audio.values_list("id", flat=True)),
+            "images": list(entry.related_images.values_list("id", flat=True)),
+            "videos": list(entry.related_videos.values_list("id", flat=True)),
+        }
+
+        source_entry_id = entry.id
+        target_entry_id = uuid.uuid4()
+        dictionary_entry_map[source_entry_id] = target_entry_id
+
+        entry.id = target_entry_id
+        entry.site = target_site
+        entry.batch_id = ""
+        entry.import_job = None
+        entry.save()
```

Comment:
Should this command pause celery signals and then reindex after all these have been created? as every entry from the copied site is being created again